R/get_zoltar_truth.R

```diff
@@ -0,0 +1,121 @@
+#' Generate versioned truth data for zoltar for a specified target
+#' It only includes national and state-level truth data.
+#' 
+#' @param target string specifying target. 
+#' Currently only support "Cumulative Deaths" and "Incident Deaths".
+#' @param issue_date optional date specifying issue date of truth data.
+#' Default to NULL which will load the latest truth data from covidData. 
+#' 
+#' @return data frame with timezero, unit, target and value
+#'
+#' @export
+preprocess_truth_for_zoltar <- function(target, issue_date = NULL){
+  
+  # validate target 
+  target <- match.arg(target, 
+                      choices = c("Cumulative Deaths", "Incident Deaths"), 
+                      several.ok = FALSE)
+  
+  # load the most up to date weeky truth data from JHU CSSE
+  df <- covidData::load_jhu_data(issue_date = issue_date, 
+                                 spatial_resolution = c('national', 'state'),
+                                 temporal_resolution = 'weekly', 
+                                 measure = 'deaths',
+                                 replace_negatives = FALSE, 
+                                 adjustment_cases = 'none',
+                                 adjustment_method = 'none')
+  
+  # select columns needed for target
+  if (target == "Cumulative Deaths"){
+    target_var <- " wk ahead cum death"
+    df <- df %>%
+      dplyr::select(-inc) %>%
+      dplyr::rename(value = cum, unit = location)
+  } else {
+    target_var <- " wk ahead inc death"
+    df <- df %>%
+      dplyr::select(-cum) %>%
+      dplyr::rename(value = inc, unit = location)
+  }
+  
```

Comment:
Can we add a comment here describing what the calculation of df below is doing?  I think the answer is that we are constructing a data frame where for each observed value at time t, that observation is the observed value at each horizon 1 through 20 relative to the corresponding past forecast timezeros from times t-1 through t-20.